### ISSUE TITLE
no-implicit-globals doesn't obey /* exported */ declaration

### ISSUE THREAD

**Issue by mwgamble** [eslint/eslint#10066]
**Tell us about your environment**

* **ESLint Version:** v4.18.2
* **Node Version:** v6.13.0
* **yarn Version:** v1.5.1
I'm not using npm.

**What parser (default, Babel-ESLint, etc.) are you using?** default

**Please show your full configuration:**

I can't provide my full configuration, but I do have `"no-implicit-globals"` set to `"error"`, and `"env"` set to `{"browser": true}`.

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
/* exported testFunc */

/**
 * @returns {int}
 */
function testFunc() {
    "use strict";

    return 1 + 2;
}
```

```bash
node_modules/.bin/eslint -c path/to/config.json path/to/test.js
```

**What did you expect to happen?**

No lint errors to be output. If I've declared that this file exports a symbol, I would have thought it's no longer implicit.

**What actually happened? Please include the actual, raw output from ESLint.**

```
path/to/test.js
  6:1  error  Implicit global variable, assign as global property instead  no-implicit-globals

✖ 1 problem (1 error, 0 warnings)
```

**Comment by kaicataldo** [comment]
Sorry that we lost track of this. Is this still an issue? `no-implicit-globals` doesn't check for comments, so there's no way for it to know what you mean by that.

**Comment by mwgamble** [comment]
Yes, it is still an issue. If you've declared a variable outside of a closure in a file designed to be run in the browser (indicated by setting the environment to "browser"), and you've listed it in the list of exports for that file, it's not an implicit global. You've explicitly declared that it is supposed to be global.

**Comment by kaicataldo** [comment]
Thanks for the response. Can you clarify what you’re proposing here?

**Comment by kaicataldo** [comment]
@mwgamble So it sounds like you're looking for this rule to be enhanced. Are you willing to make a PR if this issue is accepted?

**Comment by kaicataldo** [comment]
@eslint/eslint-team Looks like this one fell through the cracks. Mind taking a look?
